src/Storages/HDFS/ReadBufferFromHDFS.cpp

```diff
@@ -66,8 +70,12 @@ ReadBufferFromHDFS::ReadBufferFromHDFS(const std::string & hdfs_name_,
     const Poco::Util::AbstractConfiguration & config_,
     size_t buf_size_)
     : BufferWithOwnMemory<ReadBuffer>(buf_size_)
-    , impl(std::make_unique<ReadBufferFromHDFSImpl>(hdfs_name_, config_))
 {
+    const size_t begin_of_path = hdfs_name_.find('/', hdfs_name_.find("//") + 2);
```

Comment:
Need to add more checks.
Check that '//' has been found.
Check that the next '/' has been found.

Or use Poco::URI for this task?
The file name with spaces should be URL-encoded by the user.

Reply:
> Check that the next '/' has been found.

There is already  a check for user-defined url [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/HDFS/StorageHDFS.cpp#L51).

> Check that '//' has been found.

It is not necessarily present.

Reply:
If `//` is not necessarily present, what does `hdfs_name_.find("//") + 2` mean?

Reply:
> There is already a check for user-defined url here.

But it means you can use Poco::URI in this place also.